snow/engine/avalanche/transitive.go

```diff
@@ -462,28 +468,57 @@ func (t *Transitive) issue(vtx avalanche.Vertex, updatedEpoch bool) error {
 		trIDs.Add(tx.Transition().ID())
 	}
 
+	epoch, err := vtx.Epoch()
+	if err != nil {
+		return err
+	}
 	for _, tx := range txs {
-		tr := tx.Transition()
-		for _, depID := range tr.Dependencies() {
-			if !trIDs.Contains(depID) && !t.Consensus.TransitionProcessing(depID) {
+		// Mark as unfulfilled all of the transition's dependencies
+		// that are neither accepted nor processing in this tx's epoch.
+		for _, depID := range tx.Transition().Dependencies() {
+			if trIDs.Contains(depID) {
+				// This vertex contains this dependency. No need to mark
+				// as unfulfilled.
+				continue
+			}
+
+			// [processingDepTxs] is a list of processing txs that contain
+			// dependency [depID].
+			processingDepTxs := t.Consensus.ProcessingTxs(depID)
+			if len(processingDepTxs) == 0 {
+				// We don't have any processing txs with this dependency
```

Comment:
I think that `missingTransitions` is going to need to be modified to include specific epochs... Otherwise there could be a memory leak in the `trDeps` right?